python/cudf/cudf/testing/_utils.py

```diff
@@ -346,6 +346,11 @@ def get_ptr(x) -> int:
     assert len(lhs.base_children) == len(rhs.base_children)
     for lhs_child, rhs_child in zip(lhs.base_children, rhs.base_children):
         assert_column_memory_eq(lhs_child, rhs_child)
+    if isinstance(lhs, cudf.core.column.CategoricalColumn) and isinstance(
+        rhs, cudf.core.column.CategoricalColumn
+    ):
+        assert_column_memory_eq(lhs.categories, rhs.categories)
+        assert_column_memory_eq(lhs.codes, rhs.codes)
```

Comment:
Decorate this function with `@acquire_spill_lock` while we're here.

Reply:
Done

Reply:
Is this really necessary? The decorator is just an optimization (otherwise anything accessed internally just becomes permanently exposed) and it doesn't seem necessary in a testing utility. Is there a particular reason to do it here?

Reply:
> The decorator is just an optimization (otherwise anything accessed internally just becomes permanently exposed)

That's correct. 

I don't have any strong opinions here since this is a testing utility, let's see what @wence- feels about it.

Reply:
I don't really mind one way or the other.

Reply:
Thanks, dropped the spill lock here.